src/context-providers/hook.ts

```diff
@@ -0,0 +1,11 @@
+import { useContext } from "react";
+import { MainContext } from "./provider";
+
+export const useMainContext = () => {
+    const { state, dispatch } = useContext(MainContext);
+
+    return {
+        state,
+        dispatch,
```

Comment:
Not necessary to return `dispatch` since we know for sure we won't be using it like that.